sc2/bot_ai.py

```diff
@@ -715,44 +753,38 @@ def _prepare_step(self, state):
             await self.on_building_construction_complete(unit)
 
     async def _issue_unit_dead_events(self):
+        """ Its called when a building or unit is killed"""
         for unit_tag in self.state.dead_units:
             await self.on_unit_destroyed(unit_tag)
 
     async def on_unit_destroyed(self, unit_tag):
-        """ Override this in your bot class. """
-        pass
+        """ Override this in your bot class. Returns the tag of all destroyed units on that step"""
```

Comment:
This function doesn't return anything.
```suggestion
        """ Override this in your bot class. Called once for each unit destroyed in a time step with the tag of
         the destroyed unit. """
```